baseline/sweep

```diff
@@ -42,41 +54,92 @@ train() {
-                --systems "${OUTPUT}/${LEVEL}/${LG}/"*"/dev_beam${BEAM_WIDTH}.predictions" \
-                --output="${OUTPUT}/${LEVEL}/${LG}/ensemble"
+        for TRAIN in "${DATA}/${LEVEL}/"*"_train.tsv"; do
+            LG="$(basename ${TRAIN} _train.tsv)"
+            if [ ${LANGS} ] && ! [[ " ${LANGS} " =~ " ${LG} " ]]
+                # If we've passed target languages, we wish to ignore all non-targets.
+                then
+                    continue
+            fi
+            OUTPUTPATH="${OUTPUT}/${LEVEL}/${LG}/ensemble"
+            if  [ -d ${OUTPUTPATH} ] && [ "$(ls -A ${OUTPUTPATH})" ] && ! [ ${LANGS} ]
+                # We've already trained on this data up to this ensemble. So no need to do again unless we're looking for a specific language.
+                then
+                    continue
+            fi
+            for SPLIT in dev test; do
+                    python trans/ensembling.py \
+                        --gold="${DATA}/${LEVEL}/${LG}_${SPLIT}.tsv" \
+                        --systems "${OUTPUT}/${LEVEL}/${LG}/"*"/${SPLIT}_beam${BEAM_WIDTH}.predictions" \
+                        --output="${OUTPUTPATH}"
+	        done
         done
     done
 }
 
 evaluate() {
+    # Creates two-column TSV with gold and hypothesis data.
     for LEVEL in low medium high; do
-        # TODO(kbg): Revert to test path when test data becomes available.
-        for DEV in "${DATA}/${LEVEL}/"*"_dev.tsv"; do
-            LG="$(basename ${DEV} _dev.tsv)"
-            paste \
-                "${DEV}" \
-                "${OUTPUT}/${LEVEL}/${LG}/ensemble/dev_${MAX_ENSEMBLE_SIZE}ensemble.predictions" \
-            | cut -f2,4 \
-            > "${OUTPUT}/${LEVEL}/${LG}/ensemble/dev_${MAX_ENSEMBLE_SIZE}ensemble.tsv"
+        for TRAIN in "${DATA}/${LEVEL}/"*"_train.tsv"; do
+            LG="$(basename ${TRAIN} _train.tsv)"
+            for SPLIT in dev test; do
+                paste \
+                "${TRAIN//_train.tsv/_${SPLIT}.tsv}" \
+                    "${OUTPUT}/${LEVEL}/${LG}/ensemble/${SPLIT}_${MAX_ENSEMBLE_SIZE}ensemble.predictions" \
+                    | cut -f2,4 \
+                > "${OUTPUT}/${LEVEL}/${LG}/ensemble/${SPLIT}_${MAX_ENSEMBLE_SIZE}ensemble.tsv"
+            done
+        done
+    done
+    
+    # Calls evaluation script.
+    for SPLIT in dev test; do
+        echo "Split: ${SPLIT}"
+        for LEVEL in low medium high; do
+            echo "Level: ${LEVEL}"
+                python ${EVALUATION}/./evaluate_all.py \
+                    "${OUTPUT}/"*"/"*"/ensemble/${SPLIT}_${MAX_ENSEMBLE_SIZE}ensemble.tsv"
+            echo
         done
-	echo "Level: ${LEVEL}"
-        python ${EVALUATION}/./evaluate_all.py \
-            "${OUTPUT}/${LEVEL}/"*"/ensemble/dev_${MAX_ENSEMBLE_SIZE}ensemble.tsv"
-	echo
     done
 }
 
 main() {
-    rm -rf ${OUTPUT}
-    mkdir "${OUTPUT}"
     train
     ensemble
     evaluate
 }
 
+while getopts "rt:" option; do
```

Comment:
Shouldn't this go in `main`?

Reply:
mmm, I try to keep file management and set-up outside of main just as a conceptual thing personally. makes more sense that we want main focusing on the training and everything outside to just be setting things up in the directory. i'll default to you though.

Reply:
I'm okay with it if you like it.